card/client.go

```diff
@@ -166,14 +182,20 @@ func (c Client) List(params *stripe.CardListParams) *Iter {
 	var lp *stripe.ListParams
 	var p *stripe.Params
 
-	form.AppendTo(body, params)
-	lp = &params.ListParams
-	p = params.ToParams()
+	if params != nil {
+		form.AppendTo(body, params)
+		lp = &params.ListParams
+		p = params.ToParams()
+	}
 
 	return &Iter{stripe.GetIter(lp, body, func(b *form.Values) ([]interface{}, stripe.ListMeta, error) {
 		list := &stripe.CardList{}
 		var err error
 
+		if params == nil {
```

Comment:
Why not just error above like you do in the other functions `if params == nil`?

Reply:
Ah, basically because `List` functions only return an `*Iter` (and not an error). The first opportunity we get to return an error is the first time the iterator iterates, and so the check on `params` is further down than you'd expect.

Reply:
ah gotcha, that makes sense, thanks for clarifying!